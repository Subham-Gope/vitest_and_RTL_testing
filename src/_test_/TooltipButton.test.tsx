// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import TooltipButton from "../components/TooltipButton";

// describe("Testing the TooltipButton.tsx file", () => {
//   test("reveal content on hover", async () => {
//     // setting up the userEvent
//     const user = userEvent.setup();

//     // setting the debugger
//     screen.debug();

//     const { container } = render(<TooltipButton />);
//     expect(container).toMatchSnapshot();

//     const btn = screen.getByRole("button", { name: /^hover me$/i });
//     expect(btn).toBeInTheDocument();

//     // intiating the user interaction
//     await user.hover(btn);

//     const hoverText = screen.getByText(/hello there/i);
//     expect(hoverText).toBeInTheDocument();

//     await user.unhover(btn);
//     expect(screen.getByText(/\bhello there\b/i)).not.toBeInTheDocument();
//   });
// });
