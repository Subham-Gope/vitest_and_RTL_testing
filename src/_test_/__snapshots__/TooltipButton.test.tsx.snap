// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the TooltipButton.tsx file > reveal content on hover 1`] = `
<div>
  <div>
    <button
      class="px-12 py-4 bg-green-600 text-white text-3xl font-bold"
    >
      Hover Me
    </button>
  </div>
</div>
`;
